dsc/workflows/base/__init__.py

```diff
@@ -425,6 +455,60 @@ def get_bitstream_s3_uris(self, item_identifier: str) -> list[str]:
             item_identifier: The identifier used for locating the item's bitstreams.
         """
 
+    @final
+    def allow_submission(self, item_identifier: str) -> bool:
+        """Verify that a submission message should be sent based on status in DynamoDB."""
+        item_submission_record = ItemSubmissionDB.get(
+            hash_key=self.batch_id, range_key=item_identifier
+        )
```

Comment:
Are we assuming that a row will exist here, created by the reconcile step?  

If for some reason it does not exist, e.g. reconciliation didn't happen, what should we do?

Reply:
Would it make sense to use `.get_or_create()` and just kind of ignore that reconciliation didn't happen?  or do we want to bail immediately if a row doesn't exist?

Reply:
I agree that a row shouldn't be created by `submit_items` given that it wouldn't have been successfully reconciled.

> * logging a warning that the item was not found in DynamoDB
> * including it in the "skipped" count
> * then `continue`

This all seems sound to me, I'll add it!